readthedocs/search/parse_json.py

```diff
@@ -241,11 +241,10 @@ def process_mkdocs_index_file(json_storage_path, page):
                 'title': title,
                 'domain_data': {},
             })
-        else:
-            page_data.setdefault('sections', []).append({
-                'id': fragment,
-                'title': title,
-                'content': content,
-            })
+        page_data.setdefault('sections', []).append({
```

Comment:
This could use a comment explaining why this is here. The logic seems a bit odd, checking `if not fragment`, and then a default that always contains a fragment?